memgpt/server/server.py

```diff
@@ -826,8 +830,13 @@ def get_agent_messages(self, user_id: uuid.UUID, agent_id: uuid.UUID, start: int
             # return messages in reverse chronological order
             messages = sorted(page, key=lambda x: x.created_at, reverse=True)
 
-        # convert to json
-        json_messages = [record.to_json() for record in messages]
+            # Convert to json
+            # Add a tag indicating in-context or not
+            json_messages = [record.to_json() for record in messages]
+            in_context_message_ids = [str(m.id) for m in memgpt_agent._messages]
+            for d in json_messages:
```

Comment:
I didn't have any agents with enough messages to test this naturally so I tested it by setting the above clause to `False and ...` to force the `else` and it seemed to work fine.

Reply:
@sarahwooders ^